docs/design/proposals/rgd-ownership-and-deletion.md

```diff
@@ -0,0 +1,203 @@
+# ResourceGraphDefinition Ownership and Deletion Protection
+
+## Problem statement
+
+Previously, **CustomResourceDefinitions (CRDs)** generated by **ResourceGraphDefinitions (RGDs)** were only linked to their parent by a label (`metadata.labels["kro.run/resourcegraphdefinition"]`).
+There was **no ownership relation** (`ownerReferences`) or **controller reference**, leading to several issues:
+
+* The KRO controller required explicit logic to decide when to delete CRDs, controlled by a Helm flag `allowCRDDeletion`.
+* When this flag is set to `false` (the default), deleting an RGD **did not delete** its generated CRD, leaving orphaned definitions.
+* Users could still manually delete RGDs without realizing that this orphaned affected CRDs and controller behavior, causing potential **inconsistent API states**.
+
+The lack of Kubernetes-native ownership semantics made RGD–CRD lifecycle management unreliable and opaque, relying on our custom management.
+
+## Proposal
+
+Introduce proper **Kubernetes ownership semantics** between RGDs and their generated CRDs.
+Combine this with a **safe-deletion mechanism** for user-initiated RGD removals and **clearer internal deletion control** for KRO.
+
+### Overview
+
+This proposal introduces three coordinated changes:
+
+1. **Ownership introduction** — CRDs now include an `OwnerReference` pointing to the creating RGD.
+2. **Deletion protection** — A `ValidatingAdmissionPolicy` prevents accidental user deletion of RGDs.
```

Comment:
I can understand protecting against the deletion of the Kinds managed by RGDs, but what's the thinking behind stopping a user that says `k delete rgd foo` from deleting the rgd foo. Delete means delete, right?

It strikes me that this could be something that organizations choose to opt in more broadly (deletion protection for any k8s resource), but odd that KRO would take the opinion for users when other projects don't.

Reply:
Indeed, k8s standard leans towards no protection and delete everything, cascade.
Moreover, once marked for deletion, there is no way back, you must complete deletion before being able to re-use it again (`kubectl delete` starts by setting the deletionTimestamp that can't be undone)

As there is a clear 1-1 ownership link RGD <-> CRD I would indeed favour the standard behaviour

Reply:
>  I personally favor this too but adjusted after community call feedback

I think this is the kind of thing where you can easily find voices in both directions. You'll see this debate surface through many AWS APIs, too. Just try deleting an S3 bucket with objects in it.

Given the controversy and the fact that it's broader than just our API, I'd just stick with the existing standard and let a separate solution handle deletion protection holistically.